tests/test_main.py

```diff
@@ -199,3 +314,95 @@ def test_delete_project_(mock_db: MagicMock, mock_project: models.Projects) -> N
     delete_project_(mock_project, mock_db)
     mock_db.delete.assert_called_once_with(mock_project)
     mock_db.commit.assert_called_once()
+
+
+def test_create_user(mock_db: MagicMock, user_data: User) -> None:
+    create_user_(mock_db, user_data)
+    mock_db.add.assert_called_once()
+    mock_db.commit.assert_called_once()
+
+
+def test_authenticate_user_success(mock_db: MagicMock, mock_user: models.Users) -> None:
+    h = hashlib.new("SHA256")
+    h.update("test_password".encode())
+    password_hash = h.hexdigest()
+    mock_user.hashed_password = password_hash
+    mock_db.query.return_value.filter.return_value.one_or_none.return_value = mock_user
+    result = authenticate_user(mock_user.email, "test_password", mock_db)
+
+    assert result == mock_user
+
+
+def test_authenticate_user_invalid_credentials(mock_db: MagicMock, mock_user: models.Users) -> None:
+    h = hashlib.new("SHA256")
+    h.update("test_password".encode())
+    password_hash = h.hexdigest()
+    mock_user.hashed_password = password_hash
+    mock_db.query.return_value.filter.return_value.one_or_none.return_value = mock_user
+    result = authenticate_user(mock_user.email, "wrong_password", mock_db)
+    assert result is None
+
+
+def test_authenticate_user_not_found(mock_db: MagicMock) -> None:
+    mock_db.query.return_value.filter.return_value.first.return_value = None
+    result = authenticate_user("nonexistent@example.com", "password", mock_db)
+
+    assert result is None
+
+
+def test_is_project_admin_true(mock_db: MagicMock, mock_project: models.Projects, mock_user: models.Users) -> None:
+    project_id = mock_project.id
+    user_id = mock_user.id
+    mock_db.execute.return_value.one.return_value = True
+    result = is_project_admin(mock_db, project_id, user_id)
+
+    assert result is True
+
+
+def test_is_project_admin_false(mock_db: MagicMock, mock_project: models.Projects, mock_user: models.Users) -> None:
+    project_id = mock_project.id
+    user_id = mock_user.id
+    mock_db.execute.return_value.one_or_none.return_value = None
+    result = is_project_admin(mock_db, project_id, user_id)
+
+    assert result is False
+
+
+def test_get_user_existing_user(mock_db: MagicMock) -> None:
+    mock_user = MagicMock()
+    mock_db.query().filter().one_or_none.return_value = mock_user
+    result = get_user("existing@example.com", mock_db)
+
+    assert result == mock_user
+
+
+def test_get_user_non_existing_user(mock_db: MagicMock) -> None:
+    mock_db.query().filter().one_or_none.return_value = None
+    result = get_user("nonexistent@example.com", mock_db)
+
+    assert result is None
+
+
+def test_check_user_on_project_user_already_in_project(mock_db: MagicMock, mock_user: models.Users) -> None:
+    project_id = uuid.uuid4()
+    mock_existing_project = MagicMock()
+    mock_db.query().filter().one.return_value = mock_existing_project
+    result = check_user_on_project(mock_user, project_id, mock_db)
+
+    assert result is True
```

Comment:
don't check for booleans like that. Just assert result. Apply for all of the boolean checks